### ISSUE TITLE
Think about possible player actions

### ISSUE THREAD

**Issue by GambuzX** [NIAEFEUP/genetic-fighting-game#4]
And associated animations on Mixamo (https://www.mixamo.com/#/?page=1&query=punch&type=Motion%2CMotionPack)

**Comment by PedroJSilva2001** [comment]
I was thinking of implementing Mortal Kombat-like  moves (reference to MKX or MK11), mainly the blocking mechanics: 
Ex 1 - When someone goes for an overhead attack if we are crouched, we a receive a high stagger, if we are standing up, we would also be receiving but in a less pronounced way. You can't try to block it so the only way to avoid it would be to do a backstep (or jump backwards). 
Ex 2 - A low move such as an uppercut, leg sweep or low kick can only be defended when blocking in crouched position, otherwise you get staggered.
Ex 3 - A medium move like a simple hook or punch can be blocked by defending while crouching or standing up. Some could also be avoided by backsteping.
Ex 4 - A high move can be completely avoided by just crouching, you don't even need to defend.
Jumping moves would be difficult to implement I think. We could try but as it is we have enough to display in the gameplay part.
Mixamo has a lot of animations to make this work.

**Comment by GambuzX** [comment]
That may be interesting, but in terms of player actions I think this should be something simple.
I'm worried it may get a bit complex if we decide to implement different types of attacks  for each position (standing, crouching)

**Comment by GambuzX** [comment]
Can you try to detail all the moves you are thinking the player should have? Things like jump, crouch, guard, backstep, attack, crouch attack, ...
So we can better understand how complex this would be

**Comment by PedroJSilva2001** [comment]
The jumping part can be a little more difficult. 
The moves would be:
**- Normal walking:** self explanatory. Maybe we should keep this game 2d the same way MKX and MK11 are, allowing only backwards and forwards movement. 
**- Running:** self explanatory. You can't run backwards.
**- Blocking:** The character joins his arms (or whatever) to defend. Can only be done while crouched or standing up .
**- Backstep:** similar to walking backwards but faster. This doesn't need to be present since mortal kombat doesn't have them
**- Crouch:** The character goes to a low stance.

Basic Attacks:
**- Front-Punch:**
**- Back-Punch:** An uppercut is just pressing this while crouching.
**- Front-Kick:**
**- Back-Punch:**
All of these can be performed while crouching. We could also make them combo like in mkx but I don't know how complex it would be

This video might explain better what I'm trying to say:
https://www.youtube.com/watch?v=uibngKGhgxo

**Comment by GambuzX** [comment]
This looks good!

Why do you say the jumping can be more difficult? Because of finding animations for that? Maybe we can just include the act of jumping up, and not attack or block from that position

In order to simplify this on a first instance, these are my thoughts:
- Running: maybe. If this does not introduce complexity in the Genetic Algorithm it may make sense. If it somehow makes it harder I don't think is super important for now.
- Backstep: I wouldn't include this. This would allow the player to go backwards and we may get strange behaviours from the GA like infinite backsteps and such. This way it would take longer for the player to "die" and go to the next generation
- Back-Punch: How is this different from the Front-Punch? If it does not introduce something difference I don't think its important for now. Is this front / back punch system like the light attack / heavy attack, where heavy attack is slower but breaks guard?
- Combos: yeah, definitely overkill for now 😅 

The rest I think makes sense. 

Just make sure there are animations for everything.

**Comment by GambuzX** [comment]
As I mentioned on slack I think this may be a bit dependant on the theme tho

For example, if we have swords against swords it could be cool to include parrying or smt

**Comment by PedroJSilva2001** [comment]
Yes I think the part of finding animations for blocking or attacking while jumping would be the hardest part.
We could either remove back-punch and back-kick because they don't differ much from the standard attacks. They work on a fighting game perspective to increase the amount of combos each character has. Or we could make the back-punch be always a medium attack, and a normal punch a high attack. Back-kick could be a low attack and a normal kick could be a high/medium attack. I like the sword/parry idea. This all depends on the game we are trying to make.

**Comment by GambuzX** [comment]
That makes sense.
Then lets just include normal jumping without block or attack. Even crouch could just be a simple crouch.

I am not sure how much more complex having many attack moves can be, but we can just try to start with a simple set of moves (1 or 2) and then see if we can add more.

Can you synthetize the player actions on the Readme and create a PR? And include the animations you think we should use, even if only in the future

**Comment by PedroJSilva2001** [comment]
So, do I include all moves talked in my last comment + jump + sword/parry?

**Comment by GambuzX** [comment]
You can skip the sword / parry, don't think we will use it

**Comment by PedroJSilva2001** [comment]
Ok so I will write the readme with the player actions + animations. Is there a way to have a link to the individual animations on mixamo? I haven't found a way to do that.

**Comment by GambuzX** [comment]
You can just download and add them to the repository under the Assets

**Comment by PedroJSilva2001** [comment]
So I can just say "jump: jumpAnimation.extension" in the readme?

**Comment by GambuzX** [comment]
The most important thing is to document the actions. But yeah, you can reference the specific animation files that way
